### PR TITLE
feat(automation): Add custom allowlist.json

### PR DESCRIPTION
**Pull request by rajuljha**
## Description

Read custom allowlist passed by the user and populate the allowlist dict using it. Deprecation warning is printed if project is using whitelist.json.

### Changes

* Add --allowlist-path CLI argument.
* Add deprecation warning if `whitelist.json` is being used.

## How to test

Build new docker image using `Dockerfile.ci` file.

CC @GMishx @shaheemazmalmmd @Kaushl2208 @avinal

### CODE REVIEW

File: utils/automation/.github-workflow.yml [hunk 1]
```diff
@@ -11,25 +11,67 @@ jobs:
...
+          docker run --rm --name "fossologyscanner" -w "/opt/repo" -v ${PWD}:/opt/repo \
+            -e GITHUB_TOKEN=${{ github.token }} \
+            -e GITHUB_PULL_REQUEST=${{ github.event.number }} \
+            -e GITHUB_REPOSITORY=${{ github.repository }} \
             -e GITHUB_API=${{ github.api_url }} \
+            -e GITHUB_REPO_URL=${{ github.repositoryUrl }} \
+            -e GITHUB_REPO_OWNER=${{ github.repository_owner }} \
             -e GITHUB_ACTIONS \
-            fossology/fossology:scanner "/bin/fossologyscanner" nomos ojo
+            fossology/fossology:scanner "/bin/fossologyscanner" nomos ojo \
```
Comment (GMishx) [inline utils/automation/.github-workflow.yml:46]:
```suggestion
            fossology/fossology:scanner "/bin/fossologyscanner" nomos ojo
```

File: utils/automation/.github-workflow.yml [hunk 2]
```diff
@@ -38,4 +80,4 @@ jobs:
             -e GITHUB_REPO_URL=${{ github.repositoryUrl }} \
             -e GITHUB_REPO_OWNER=${{ github.repository_owner }} \
             -e GITHUB_ACTIONS \
-            fossology/fossology:scanner "/bin/fossologyscanner" copyright keyword
+            fossology/fossology:scanner "/bin/fossologyscanner" copyright keyword \
```
Comment (GMishx) [inline utils/automation/.github-workflow.yml:83]:
```suggestion
            fossology/fossology:scanner "/bin/fossologyscanner" copyright keyword
```

File: utils/automation/fossologyscanner.py [hunk 3]
```diff
@@ -67,16 +68,23 @@ def get_api_config() -> ApiConfig:
...
+    if os.path.exists('whitelist.json'):
+      file_name = 'whitelist.json'
+      print("Reading whitelist.json file...")
+      logging.warning("Name 'whitelist.json' is deprecated. Please use 'allowlist.json instead'")
+    else:
+      file_name = 'allowlist.json'
+      print("Reading allowlist.json file...")
   else:
-    file_name = 'allowlist.json'
+      file_name = path
```
Comment (GMishx) [inline utils/automation/fossologyscanner.py:87]:
```suggestion
    file_name = path
```

**Review by GMishx — CHANGES_REQUESTED** [review]
Minor changes needed.

**Review by GMishx — APPROVED** [review]
Changes looks good.

File: utils/automation/.github-workflow.yml [hunk 4]
```diff
@@ -11,25 +11,67 @@ jobs:
...
-    steps:
-      - uses: actions/checkout@v3
-      - run: |
+      - name: Checkout this repository
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Run Scanners (excluding allowlisted)
```
Comment (Kaushl2208) [inline utils/automation/.github-workflow.yml:20]:
```suggestion
      - name: Run Scanners (including allowlisted)
```

**Review by Kaushl2208 — CHANGES_REQUESTED** [review]

File: utils/automation/.github-workflow.yml [hunk 5]
```diff
@@ -11,25 +11,67 @@ jobs:
   check-license:
     name: Check license
     runs-on: ubuntu-22.04
-    steps:
```
[the remark this answers is not in the document]
Comment (rajuljha) [reply utils/automation/.github-workflow.yml:14]: Corrected

File: utils/automation/.github-workflow.yml (hunk 4 above)
Comment (rajuljha) [reply utils/automation/.github-workflow.yml:20]: `Excluding` allowlisted I thought would mean excluding the allowlisted licenses from scan. Should I change it still? It's probably a grammatical issue 🤓

Comment (Kaushl2208) [reply utils/automation/.github-workflow.yml:20]: Yea yea, I think I got it in another reference. Maybe (Run Scanners) is just fine.  Like in the Copyright stage

**Review by Kaushl2208 — APPROVED** [review]
Changes looks good! @rajuljha 
Needs test!

**Review by Kaushl2208 — APPROVED** [review]
Tested, working as expected

**Pull request merged**